upgrade/upgrade-provider.go

```diff
@@ -89,6 +90,11 @@ func UpgradeProvider(ctx Context, name string) error {
 
 				var previous string
 				if repo.currentUpstreamVersion != nil {
+					if semver.Compare(repo.currentUpstreamVersion.String(),
+						upgradeTargets.Latest().String()) > 0 {
```

Comment:
I wonder if we could/should filter these out in [GetExpectedTargets](https://github.com/pulumi/upgrade-provider/blob/main/upgrade/steps.go#L496)?

Reply:
`GetExpectedTargets` fetches the latest version (<= `--provider-version` if set), and then we get the current version from `setCurrentUpstream*` 
I think we can keep it here for now, but I think we should add a separate function to directly fetch the upstream version vs looking at issues?

Reply:
I'm fine keeping it here, you make a good point about needing a separate function that bypasses the issue crawling.